pkg/kubelet/dockershim/network/kubenet/kubenet_linux.go

```diff
@@ -464,44 +588,73 @@ func (plugin *kubenetNetworkPlugin) TearDownPod(namespace string, name string, i
 		return fmt.Errorf("Kubenet needs a PodCIDR to tear down pods")
 	}
 
-	// no cached IP is Ok during teardown
-	podIP, _ := plugin.podIPs[id]
-	if err := plugin.teardown(namespace, name, id, podIP); err != nil {
+	if err := plugin.teardown(namespace, name, id); err != nil {
 		return err
 	}
 
 	// Need to SNAT outbound traffic from cluster
 	if err := plugin.ensureMasqRule(); err != nil {
 		klog.Errorf("Failed to ensure MASQ rule: %v", err)
 	}
-
 	return nil
 }
 
 // TODO: Use the addToNetwork function to obtain the IP of the Pod. That will assume idempotent ADD call to the plugin.
 // Also fix the runtime's call to Status function to be done only in the case that the IP is lost, no need to do periodic calls
 func (plugin *kubenetNetworkPlugin) GetPodNetworkStatus(namespace string, name string, id kubecontainer.ContainerID) (*network.PodNetworkStatus, error) {
-	plugin.mu.Lock()
-	defer plugin.mu.Unlock()
-	// Assuming the ip of pod does not change. Try to retrieve ip from kubenet map first.
-	if podIP, ok := plugin.podIPs[id]; ok {
-		return &network.PodNetworkStatus{IP: net.ParseIP(podIP)}, nil
+	// try cached version
+	networkStatus := plugin.getNetworkStatus(id)
+	if nil != networkStatus {
+		return networkStatus, nil
 	}
 
+	// not a cached version, get via network ns
 	netnsPath, err := plugin.host.GetNetNS(id.ID)
 	if err != nil {
 		return nil, fmt.Errorf("Kubenet failed to retrieve network namespace path: %v", err)
 	}
 	if netnsPath == "" {
 		return nil, fmt.Errorf("Cannot find the network namespace, skipping pod network status for container %q", id)
 	}
-	ip, err := network.GetPodIP(plugin.execer, plugin.nsenterPath, netnsPath, network.DefaultInterfaceName)
+	ips, err := network.GetPodIPs(plugin.execer, plugin.nsenterPath, netnsPath, network.DefaultInterfaceName)
 	if err != nil {
 		return nil, err
 	}
 
-	plugin.podIPs[id] = ip.String()
-	return &network.PodNetworkStatus{IP: ip}, nil
+	// cache the ips
+	for _, ip := range ips {
+		plugin.addPodIP(id, ip.String())
+	}
+
+	// return from cached
+	return plugin.getNetworkStatus(id), nil
+}
+
+// returns networkstatus
+func (plugin *kubenetNetworkPlugin) getNetworkStatus(id kubecontainer.ContainerID) *network.PodNetworkStatus {
+	plugin.mu.Lock()
+	defer plugin.mu.Unlock()
+
+	// Assuming the ip of pod does not change. Try to retrieve ip from kubenet map first.
+	iplist, ok := plugin.podIPs[id]
+	if !ok {
+		return nil
+	}
+	// sort making v4 first
+	ips := make([]net.IP, 0)
```

Comment:
move this before the first return so the return type is consistent

Reply:
The return type is `*network.PodNetworkStatus` and is consistent either an `error` or a valid type.